src/test/java/com/commercetools/project/sync/state/StateSyncerTest.java

```diff
@@ -0,0 +1,79 @@
+package com.commercetools.project.sync.state;
+
+import com.commercetools.sync.states.StateSync;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.states.State;
+import io.sphere.sdk.states.StateDraft;
+import io.sphere.sdk.states.StateDraftBuilder;
+import io.sphere.sdk.states.StateDraftDsl;
+import io.sphere.sdk.states.StateType;
+import io.sphere.sdk.states.expansion.StateExpansionModel;
+import io.sphere.sdk.states.queries.StateQuery;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import static com.commercetools.project.sync.util.TestUtils.getMockedClock;
+import static io.sphere.sdk.json.SphereJsonUtils.readObjectFromResource;
+import static java.util.Arrays.asList;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+class StateSyncerTest {
+
+    @Test
+    void of_ShouldCreateStateSyncerInstance() {
+        final StateSyncer stateSyncer
+                = StateSyncer.of(mock(SphereClient.class), mock(SphereClient.class), getMockedClock());
+
+        // assertions
+        assertThat(stateSyncer).isNotNull();
+        assertThat(stateSyncer.getQuery()).isEqualTo(StateQuery.of());
+        assertThat(stateSyncer.getSync()).isInstanceOf(StateSync.class);
+    }
+
+    @Test
+    void transform_ShouldReplaceReferenceIdsWithKeys() {
+        final StateSyncer stateSyncer
+                = StateSyncer.of(mock(SphereClient.class), mock(SphereClient.class), getMockedClock());
+        final List<State> states =
+                asList(
+                        readObjectFromResource("state-1.json", State.class),
+                        readObjectFromResource("state-2.json", State.class));
+
+        final CompletionStage<List<StateDraft>> stateDrafts = stateSyncer.transform(states);
```

Comment:
should not we use r`eplaceStateTransitionIdsWithKeys` instead of creating the drafts manually ?

Reply:
This is done in other tests and that's why we get the issue: https://github.com/commercetools/commercetools-project-sync/issues/129. I think it's better to create explicitly what you expect.

Reply:
I would not mention this as a bug but rather than a missing obj in the type reference of the sample.
As explained before, 
```
It looks you are right, the test scenario is not useless but not prepared correctly. In the transform method, it’s calling the replacement util, reference is custom type. But there is no key in reference expanded in JSON, so we need to add expanded obj there.
```

Here in-state sample that you have created is correct, so you need to use reference expansion utility rather than creating an object builder, as your code and the util are doing the same.

Reply:
This test worked in my local, you could skip this comment if you prefer to write it explicitly..

```java
@Test
  void transform_ShouldReplaceReferenceIdsWithKeys() {
    final StateSyncer stateSyncer =
        StateSyncer.of(mock(SphereClient.class), mock(SphereClient.class), getMockedClock());
    final List<State> states =
        asList(
            readObjectFromResource("state-1.json", State.class),
            readObjectFromResource("state-2.json", State.class));

    final CompletionStage<List<StateDraft>> stateDrafts = stateSyncer.transform(states);
    // assertions
    final List<StateDraft> expectedResult = replaceStateTransitionIdsWithKeys(states);
    assertThat(stateDrafts).isCompletedWithValue(expectedResult);
  }
```

Reply:
Can you change the name to 

transform_ShouldReplaceReferenceIdsWithKeys -> transform_ShouldReplaceStateTransitionIdsWithKeys

Reply:
In tests I try to avoid copy the implementation if it's possible. In this case I would rather really compare the result. Let's say now there's a bug and it suddenly stops expanding transitions in java sync. If I use the util method exactly like in the implementation, I would not know there's an error as everything would pass. Or if someone remove the expansion in the `state-1.json` file, I would also not know as test would pass again.

Reply:
> Can you change the name to
> 
> transform_ShouldReplaceReferenceIdsWithKeys -> transform_ShouldReplaceStateTransitionIdsWithKeys

Thanks :) I thought I changed, but somehow it's not there. https://github.com/commercetools/commercetools-project-sync/pull/128/commits/d55ca851bb235c756e7b46871312ab618211b4e9